src/utils/fritekstUtils.ts

```diff
@@ -1,153 +1,180 @@
+        if (erInnholdTekstObjekt(innhold)) {
+            return sanityBlocktekstToHtml(innhold as SanityTekstObjekt);
+        } else {
+            return [];
+        }
+    });
+};
 
 export const fjernSpesialKarakterer = (str: string): string => {
-	return str
-		.replaceAll('&nbsp;', ' ')
-		.replaceAll('\n', '')
-		.replaceAll('&lt;', '')
-		.replaceAll('&gt;', '');
-}
-
-export const oppdaterFritekstTabellMedTekst =(fritekstTabellKopi: string[][], nyFritekst: string, gammelFritekstTabell: string[][]): string[][] => {
-
-        const nyFritekstSterilisert = fjernSpesialKarakterer(nyFritekst);
-        const gammelFritekst = dobbelTabellTilStreng(gammelFritekstTabell);
-        const antallTegnLagtTil = finnAntallTegnLagtTil(nyFritekstSterilisert, gammelFritekst);
-        const endringsIndeks = finnEndringsIndeks(nyFritekstSterilisert, gammelFritekst);
-
-        let karakterTeller = 0;
-        let erEndringGjort = false;
+    return str
+        .replaceAll('&nbsp;', ' ')
+        .replaceAll('\n', '')
+        .replaceAll('&lt;', '')
+        .replaceAll('&gt;', '');
+};
 
-        return fritekstTabellKopi.map((fritekstTabellElement) => {
+export const oppdaterFritekstTabellMedTekst = (
+    fritekstTabellKopi: string[][],
+    nyFritekst: string
+): { nyFritekstTabell: string[][]; flettefeltNummer?: number; nyttFritekstElement?: string } => {
+    const nyFritekstSterilisert = fjernSpesialKarakterer(nyFritekst);
+    const gammelFritekst = dobbelTabellTilStreng(fritekstTabellKopi);
+    const antallTegnLagtTil = finnAntallTegnLagtTil(nyFritekstSterilisert, gammelFritekst);
+    const endringsIndeks = finnEndringsIndeks(nyFritekstSterilisert, gammelFritekst);
+
+    let karakterTeller = 0;
+    let erEndringGjort = false;
+
+    let nyttFritekstElement;
+    let flettefeltIndeksITabell = 0;
+    let nyFritekstTabell = fritekstTabellKopi;
+    if (endringsIndeks !== -1) {
+        nyFritekstTabell = fritekstTabellKopi.map((fritekstTabellElement) => {
             const gammelKarakterTeller = karakterTeller;
             karakterTeller += finnKaraktererIListeMedStrenger(fritekstTabellElement);
 
             if (karakterTeller > endringsIndeks && !erEndringGjort) {
                 erEndringGjort = true;
-                const { tabellIndeks, nyttFritekstElement } = finnTabellIndeksOgNyttFritekstElement(
+                const tabbellIndeksOgNyttFritekstElement = finnTabellIndeksOgNyttFritekstElement(
                     endringsIndeks,
                     nyFritekstSterilisert,
                     fritekstTabellElement,
                     antallTegnLagtTil,
                     gammelKarakterTeller
                 );
-
-
-                fritekstTabellElement[tabellIndeks] = nyttFritekstElement; 
-				
+                const tabellIndeks = tabbellIndeksOgNyttFritekstElement.tabellIndeks;
+                nyttFritekstElement = tabbellIndeksOgNyttFritekstElement.nyttFritekstElement;
```

Comment:
Er det en grunn til at dette er endret fra `const {tabellIndeks, nyttFritekstElement}` til å sette de hver for seg senere? Hvis det ikke har en effekt på koden syntes jeg det er ryddigere + mindre kode å gjøre det på forrige måten

Reply:
"tabbellIndeksOgNyttFritekstElement" er skrevet feil, tabell med en b